File: backend/src/main/kotlin/cta/model/FavoriteCar.kt
package cta.model

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table
import java.time.LocalDateTime

@Entity
@Table(name = "favorite_car")
class FavoriteCar : BaseEntity() {
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "buyer_id", nullable = false)
    lateinit var buyer: Buyer

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "car_id", nullable = false)
    lateinit var car: Car

    @Column(name = "date_added", nullable = false)
    var dateAdded: LocalDateTime = LocalDateTime.now()

    @Column(name = "rating")
    var rating: Int? = null

    @Column(name = "comment", columnDefinition = "TEXT")
    var comment: String? = null

    @Column(name = "price_notifications")
    var priceNotifications: Boolean = false

    fun updateRating(newRating: Int?) {
        this.rating = newRating
    }

    fun updateComment(newComment: String?) {
        this.comment = newComment?.ifBlank { null }
    }

    fun enablePriceNotifications() {
        priceNotifications = true
    }

    fun disablePriceNotifications() {
        priceNotifications = false
    }

    fun togglePriceNotifications() {
        priceNotifications = !priceNotifications
    }

    fun isReviewed(): Boolean {
        return rating != null || !comment.isNullOrBlank()
    }
}

File: frontend/src/types/favoriteCar.ts
import { Buyer } from "./buyer";
import { Car } from "./car";

export interface FavoriteCarCreateRequest {
  buyerId: string | number;
  carId: string | number;
  dateAdded: string; // ISO format: "yyyy-MM-dd'T'HH:mm:ss"
  rating: number; // 0-10
  comment: string;
  priceNotifications?: boolean;
}

export interface FavoriteCarUpdateReviewRequest {
  rating: number;
  comment: string;
}

export interface Favorite {
  id: number | string;
  car: Car;
  buyer: Buyer;
  dateAdded: string;
  rating: number;
  comment: string | null;
  priceNotifications: boolean;
}